packages/onboard-helpers/src/hooks/useOnboardHelpers.ts

```diff
@@ -0,0 +1,356 @@
+  updateLocale: (locale: string) => void,
+  updateStorageWalletName: (walletName: string) => void,
+  updateRouterProps: () => void
+) => {
+  const [{ wallet }, connect, disconnect] = useConnectWallet()
+  const [_, setChain] = useSetChain()
+  const updateWalletLocale = useSetLocale()
+  const navigate = useNavigate()
+  const location = useLocation()
+
+  const locationPathname = location.pathname
+  const { walletChainId, walletSignerAddress } = getWallet(wallet)
+
+  const isSameChainId = {
+    apiRoute: apiChainId === routeChainId && !isIntegrationPage,
+    apiWallet: !!walletChainId && +walletChainId === apiChainId,
+    routeWallet: !!walletChainId && +walletChainId === routeChainId,
+    apiWalletRoute:
+      !!walletChainId && +walletChainId === apiChainId && +walletChainId === routeChainId && !isIntegrationPage,
+  }
+
+  const isSameAddress = {
+    apiWallet: apiSignerAddress.toLowerCase() === walletSignerAddress,
+  }
+
+  const isSameLocale = routeLocale === document.documentElement.lang
+
+  const updateConnectSwitchNetwork = useCallback(
+    (options: ConnectOptions[CONNECT_STAGE.SWITCH_NETWORK]) => {
+      updateConnectState('loading', CONNECT_STAGE.SWITCH_NETWORK, options)
+    },
+    [updateConnectState]
+  )
+
+  const updateConnectApi = useCallback(
+    (options: ConnectOptions[CONNECT_STAGE.CONNECT_API]) => {
+      updateConnectState('loading', CONNECT_STAGE.CONNECT_API, options)
+    },
+    [updateConnectState]
+  )
+
+  useEffect(() => {}, [connectState.stage, connectState.status])
+
+  const handleReconnectWallet = useCallback(
+    async (walletState: WalletState, chainId?: number) => {
+      const label = walletState.label
+      await disconnect(walletState)
+      const newWalletState = (await connect({ autoSelect: { label, disableModals: true } }))[0]
+      if (chainId) {
+        await setChain({ chainId: ethers.toQuantity(chainId) })
+        await _isNetworkSwitched(chainId)
+      }
+      return newWalletState
+    },
+    [connect, disconnect, setChain]
+  )
+
+  const handleConnectCurveApi = useCallback(
+    async ([chainId, useWallet]: ConnectOptions[CONNECT_STAGE.CONNECT_API]) => {
+      try {
+        let parsedWallet = useWallet ? wallet : null
+        const { walletChainId } = getWallet(parsedWallet)
+
+        // check wallet sync again
+        if (parsedWallet && walletChainId && +walletChainId !== chainId) {
+          parsedWallet = await handleReconnectWallet(parsedWallet, chainId)
+        }
+
+        await updateApi(chainId, useWallet ? parsedWallet : null)
+        updateConnectState('success', '')
+      } catch (error) {
+        console.error(error)
+        updateConnectState('failure', CONNECT_STAGE.CONNECT_API)
+      }
+    },
+    [wallet, updateApi, updateConnectState, handleReconnectWallet]
+  )
+
+  const handleConnectWallet = useCallback(
+    async (walletName: ConnectOptions[CONNECT_STAGE.CONNECT_WALLET]) => {
+      let walletState
+
+      if (!walletName) {
```

Comment:
Declare here a new function to get walletState and just return it
```suggestion
      const getWalletState = async () => {
        if (!walletName) return (await connect())?.[0]
      ...
```